src/Groups/homomorphisms.jl

```diff
@@ -616,6 +616,26 @@ function isomorphism(::Type{GrpAbFinGen}, A::GrpAbFinGen)
    end::AbstractAlgebra.Generic.IdentityMap{GrpAbFinGen}
 end
 
+# We need not find independent generators in order to create
+# a presentation of a fin. gen. abelian group.
+function isomorphism(::Type{FPGroup}, A::GrpAbFinGen)
+   # Known isomorphisms are cached in the attribute `:isomorphisms`.
+   isos = get_attribute!(Dict{Type, Any}, A, :isomorphisms)
+   return get!(isos, FPGroup) do
+      G = free_group(ngens(A))
+      R = rels(A)
+      s = vcat(elem_type(G)[i*j*inv(i)*inv(j) for i = gens(G) for j = gens(G) if i != j],
+           elem_type(G)[reduce(*, [gen(G, i)^R[j,i] for i=1:ngens(A) if !iszero(R[j,i])], init = one(G)) for j=1:nrows(R)])
```

Comment:
Hmmm, I wonder if there was a reason @fieker did not use `prod`?

Also, the `elem_type(G)` shouldn't be necessary?

So perhaps
```suggestion
           [prod([gen(G, i)^R[j,i] for i=1:ngens(A) if !iszero(R[j,i])], init = one(G)) for j=1:nrows(R)])
```
or even
```suggestion
           [prod(gen(G, i)^R[j,i] for i=1:ngens(A) if !iszero(R[j,i]), init = one(G)) for j=1:nrows(R)])
```
And similar for `reduce(+, ...) -> sum` ? And the other `elem_type(G)` above?

Reply:
I would guess the code predates 1.6. There was no `init` for `prod` before 1.6.

Why remove the `elem_type(G)`? The julia compiler can be very touchy (aka throwing errors) for comprehensions. I have spent days fixing type inference errors which were fixed by simply adding these types.

Reply:
O.k., I replace `reduce` by `prod` and `sum`, respectively.

Concerning `elem_type(G)`, this information is necessary if the vector in question can be empty. Are there other situations where `elem_type(G)` cannot be omitted?

Reply:
I am not talking about the type being inferred as `Vector{Any}`. I am talking about the compiler crashing because things got too complicated. This is not hypothetical, e.g., https://github.com/thofma/Hecke.jl/search?q=compiler&type=commits.